src/non_cached.rs

```diff
@@ -83,7 +84,7 @@ where
         self.max_batch_size
     }
 
-    pub async fn load(&self, key: K) -> V {
+    pub async fn load_safe(&self, key: K) -> Result<V, Error> {
```

Comment:
to align with Rust ecosystem, may rename as `try_load`?

Reply:
definitely better